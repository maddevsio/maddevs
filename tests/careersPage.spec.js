import {
  mount
} from '@vue/test-utils';
import CareersPage from '@/pages/_lang/careers';

describe('Careers page', () => {
  let wrapper;

  beforeEach(() => {
    global.$nuxt = {
      $route: {
        name: null
      }
    };
    wrapper = mount(CareersPage, {
      stubs: ['ValidationObserver']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('Sets the correctly data', () => {
    expect(wrapper.vm.$data.title).toEqual('Mad Devs Careers: Work with Us');
    expect(wrapper.vm.$data.description).toEqual('Join our team of 100+ staff members working on unique projects, master new IT skills, and build a fantastic career path in software engineering, DevOps, or mobile app development.');
    expect(wrapper.vm.$data.ogUrl).toEqual('https://maddevs.io/careers');
  });
});
