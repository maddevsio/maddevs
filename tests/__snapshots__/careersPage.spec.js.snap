// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Careers page renders correctly 1`] = `
<main
  class="main"
>
  <div
    class="container banner-container"
  >
    <div
      class="banner-content"
    >
      <div
        class="text-content_wrapper"
      >
        <div
          class="main-title-wrapper"
        >
          <h1
            class="banner-main_title--white-first"
          >
            
        Look!
      
          </h1>
           
          <span
            class="banner-main_title--white"
          >
            A new Mad Devs' ranger is here!
          </span>
        </div>
      </div>
       
      <div
        class="navigation-wrap"
      >
        <nav
          class="navigation-list"
        >
          <router-link
            class="navigation-item"
            to="services/#remote-tech-staff"
          >
            Remote Tech Staff
          </router-link>
           
          <router-link
            class="navigation-item"
            to="services/#software-development"
          >
            Software Development
          </router-link>
           
          <router-link
            class="navigation-item"
            to="services/#infrastructure-optimisation"
          >
            Infrastructure Optimization
          </router-link>
           
          <router-link
            class="navigation-item"
            to="services/#it-consulting"
          >
            IT Consulting
          </router-link>
        </nav>
      </div>
    </div>
  </div>
   
  <section
    class="careers"
    id="careers"
  >
    <div
      class="container"
    >
      <div
        class="careers__wrapper"
      >
        <img
          alt="Careers Background Image"
          class="careers__background-logo"
          src="@/assets/img/Careers/svg/careers_logo.svg"
        />
         
        <validationobserver-stub />
      </div>
    </div>
     
    <transition-stub
      name="fade"
    >
      <!---->
    </transition-stub>
  </section>
</main>
`;
